src/lib.rs

```diff
@@ -50,10 +81,12 @@ fn pid1_handling(child: Option<Child>) -> ! {
                 };
                 (|| {
                     let child = child?;
-                    let pid = pid?;
+                    let child_exit_status = pid?;
+                    let pid = child_exit_status.pid;
                     let pid = u32::try_from(pid.as_raw()).ok()?;
                     if pid == child {
-                        graceful_shutdown()
```

Comment:
Calling the abort call previously resulted in the segmentation fault issue.

Reply:
Wow, that's really interesting, I wouldn't have expected that. I think I've seen that in other code previously, not realizing that was the cause. Good catch!